src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp

```diff
@@ -97,15 +111,29 @@ void FlybyPoint::UpdateState(uint64_t CurrentSimNanos)
         this->f0 = v3Norm(relativeState.v_BN_N) / v3Norm(relativeState.r_BN_N);
 
         /*! compute flight path angle at time of read */
-        this->gamma0 = atan(v3Dot(relativeState.v_BN_N, ur_N) / v3Dot(relativeState.v_BN_N, ut_N));
+        if (this->singularityFlag == -1) {
+            this->gamma0 = -M_PI_2;
+        }
+        else if (this->singularityFlag == 1) {
+            this->gamma0 = M_PI_2;
+        }
+        else {
+            this->gamma0 = atan(v3Dot(relativeState.v_BN_N, ur_N) / v3Dot(relativeState.v_BN_N, ut_N));
+        }
 
         /*! update lastFilterReadTime to current time and dt to zero */
         this->lastFilterReadTime = CurrentSimNanos;
         dt = 0;
     }
 
     /*! compute rotation angle of reference frame from last read time */
-    double theta = atan(tan(this->gamma0) + this->f0 / cos(this->gamma0) * dt) -this->gamma0;
+    double theta;
+    if (this->singularityFlag == 0) {
+        theta = atan(tan(this->gamma0) + this->f0 / cos(this->gamma0) * dt) - this->gamma0;
+    }
+    else {
+        theta = 0;
+    }
```

Comment:
Maybe I'm wrong, but I believe this could be done inside the previous ifs as well to avoid several similar ifs

Reply:
I don't think so. This theta in particular is a function of dt, and therefore needs to be calculated outside of the if statement that starts on line 71. Otherwise it would not get updated at every call, but only at every filter read. Because I need a variable (singularityFlag) to pass that information from inside that big if statement, I think it makes more sense to leave it.

Reply:
Ok